nms/app/packages/magmalte/app/views/subscriber/SubscriberAddDialog.js

```diff
@@ -223,6 +223,103 @@ export default function AddSubscriberButton(props: {onClose: () => void}) {
   );
 }
 
+export function BulkEditSubscriberButton(props: {onClose: () => void}) {
+  const classes = useStyles();
+  const successCountRef = useRef(0);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+
+  const fileInput = useRef(null);
+  const saveSubscribers = async (subscribers: Array<SubscriberInfo>) => {
```

Comment:
In `AddSubscriberDetails` there's a very similar `saveSubscribers` function. Separate out that function and refactor it so it serves both these components